src/Storefront/Framework/Routing/RequestTransformer.php

```diff
@@ -98,8 +98,14 @@ public function transform(Request $request): Request
             throw new SalesChannelMappingException($request->getUri());
         }
 
-        $absoluteBaseUrl = $this->getSchemeAndHttpHost($request) . $request->getBaseUrl();
-        $baseUrl = str_replace($absoluteBaseUrl, '', $salesChannel['url']);
+        $originalBaseUrl = $request->getBaseUrl();
```

Comment:
Hi @hlohaus 

what exactly are you trying to achieve with this change?
If no sales channel was found to the URL, it already jumps out in line 94.

If the sales channel is incorrectly configured (e.g. if the BasePath is missing) then a corresponding error should occur.

An autocorrect of the wrong configuration would only lead to unexpected behavior on more complex systems.

Reply:
The sale channel will be found. Try out my sub dir project.
Without this change all urls in storefront are wrong.
See screenshot.
<img width="684" alt="Bildschirmfoto 2019-12-11 um 13 32 14" src="https://user-images.githubusercontent.com/983577/70621901-f72efa80-1c1a-11ea-82b3-3475a2851a4e.png">